elks/tools/mkromfs.c

```diff
@@ -225,8 +228,25 @@ static int parse_dir (inode_build_t * grand_parent_inode,
 					if (err) break;
 					}
 
+				else if (S_ISCHR (child_stat.st_mode))
```

Comment:
There is common code in these two `else if` sections; can that code be combined or must it be duplicated? Perhaps a good place to use a `goto`?

Reply:
Yes, a `goto` simplifies the code. Done for next commit.